packages/assistant/src/RPA/Assistant/library.py

```diff
@@ -0,0 +1,991 @@
+            Add heading   Delete user ${username}?
+            Add submit buttons    buttons=No,Yes    default=Yes
+            ${result}=    Run dialog
+            IF   $result.submit == "Yes"
+                Delete user    ${username}
+            END
+        """
+        buttons, default = to_options(buttons, default)
+        for button in buttons:
+            self._add_closing_button(button)
+
+    @keyword("Run dialog", tags=["dialog"])
+    def run_dialog(
+        self,
+        timeout: int = 180,
+        title: str = "Dialog",
+        height: Union[int, str] = "AUTO",
+        width: int = 480,
+        on_top: bool = False,
+    ) -> Result:
+        """Create a dialog from all the defined elements and block
+        until the user has handled it.
+
+        :param timeout: Time to wait for dialog to complete, in seconds
+        :param title:  Title of dialog
+        :param height: Height of dialog (in pixels or 'AUTO')
+        :param width:  Width of dialog (in pixels)
+        :param on_top: Show dialog always on top of other windows
+
+
+        Returns a result object with all input values.
+
+        Example:
+
+        .. code-block:: robotframework
+
+            Add heading     Please enter your username
+            Add text input  name=username
+            ${result}=      Run dialog
+            Log    The username is: ${result.username}
+        """
+
+        # FIXME: support timeout
+
+        self._client.display_flet_window(title, height, width, on_top)
+        return self._client.results
+
+    @keyword("Ask User", tags=["dialog"])
+    def ask_user(self, timeout: int = 180, **options: Any) -> Result:
+        """Same as ``Run Dialog`` it will create a dialog from all the defined
+        elements and block until the user has handled it. It will also add
+        by default a submit and close buttons.
+
+        :param timeout: Time to wait for dialog to complete, in seconds
+        :param options: Options for the dialog
+
+        Returns a result object with all input values.
+
+        For more information about possible options for opening the dialog,
+        see the documentation for the keyword ``Run Dialog``.
+
+        Example:
+
+        .. code-block:: robotframework
+
+            Add heading     Please enter your username
+            Add text input  name=username
+            ${result}=      Ask User
+            Log    The username is: ${result.username}
+        """
+
+        # FIXME: support timeout
+
+        self.add_submit_buttons(["Submit", "Close"], "Submit")
+        self._client.display_flet_window(**options)
+        return self._client.results
+
+    @keyword("Refresh", tags=["dialog"])
+    def refresh(self):
+        """Can be used to update UI elements when adding elements while dialog is running"""
+        if self._client.page:
+            self._client.update_elements(self._client.page)
+        else:
+            raise Exception("No dialog open")
```

Comment:
Don't raise base Exceptions, turn it into a RuntimeError or something

Reply:
agreed, done